README.md

```diff
@@ -43,17 +57,127 @@ Snippets for deployment template scaffolding and 70+ Azure resources.
+
+Select **Browse** from the context menu and then select the parameter file.
+
+![Image showing the parameter file creation options (none, new, and browse)](./images/create-parameter-file-2.png)
+
+### Parameter validation
+
+Once a parameter file has been associated with a template, the extension provides parameter validation across both files. In the following example, a parameter is specified with a set of allowed values. In the parameter file, a value is specified that is not part of the allowed values list. Because of this, the extension produces an error.
+
+![Image showing a parameter validation issue](./images/param-file-validation.png)
+
+### Add missing parameters
+
+To add missing parameters to an existing parameter file, ensure that an association has been created, select `parameters`, and then click the light bulb icon.
+
+![Image showing the light bulb helper for adding missing parameters](./images/missing-params-one.png)
+
+Select the option to add only required or all missing parameters.
+
+![Image showing missing parameter options (Add missing required parameters, Add all missing parameters)](./images/missing-params-two.png)
 
-## Intellisense
+Enter a value for the added parameters and also remove the 'TODO' comment.
+
+![Image showing parameter file after missing parameters have been added](./images/missing-params-three.png)
+
+### Find all parameter references
+
+Right-click on a parameter name and select **Go to References**. Double-clicking on any reference moves your cursor to the reference location.
+
+![Image showing the go to reference user interface](./images/parameter-references.png)
+
+### Rename a parameter
+
+Right-click on the parameter name and select **Rename Symbol**. This action renames the parameter and all template references to the parameter.
+
+![Image showing the rename parameter dialoug](./images/rename-parameter.png)
+
+### Remove or update parameter file association
+
+Click on the code lense option for changing a parameter association.
+
+![Image showing the code lense for creating a parameter file association](./images/undo-change-mapping.png)
+
+Select a new parameter file or create a new parameter file to update the association. Select **None** to remove the parameter file association.
+
+![Image showing the parameter file creation options (none, new, and browse)](./images/undo-change-mapping-two.png)
+
+## Template navigation
+
+The ARM tools extension for VS Code offers several features for navigating around an ARM template.
+
+### ARM template outline
+
+The ARM template outline is used to navigate to and select any element of an ARM template.
+
+![Image showing the ARM template outline with a storage account resource selected](./images/arm-template-outline.png)
+
+In addition to navigation, several features such as insert item and template sorting can be engaged using the buttons found on the ARM template outline.
+
+### Go to Definition
```

Comment:
I'd reference Find Definitions here as well.

Reply:
I've added a find all references just below. Is this what you mean by find definition?